packages/espressocash_app/lib/features/token_chart/src/data/token_chart_cache.dart

```diff
@@ -0,0 +1,67 @@
+import 'dart:convert';
+
+import 'package:drift/drift.dart';
+import 'package:injectable/injectable.dart';
+
+import '../../../../data/db/db.dart';
+import '../../../../data/db/mixins.dart';
+import '../token_chart_item.dart';
+
+const _maxAge = Duration(hours: 1);
+
+@injectable
+class TokenChartCache {
+  TokenChartCache({
+    required MyDatabase db,
+  }) : _db = db;
+
+  final MyDatabase _db;
+
+  Future<List<TokenChartItem>?> get(String id) async {
+    final query = _db.select(_db.tokenChartCacheRows)
+      ..where((p) => p.id.equals(id));
+
+    final row = await query.getSingleOrNull();
+
+    if (row != null) {
+      final now = DateTime.now();
+
+      if (now.difference(row.created) <= _maxAge) {
+        return row.toModel();
+      } else {
+        await remove(id);
+      }
+    }
+  }
+
+  Future<void> set(String id, List<TokenChartItem> result) => _db
+      .into(_db.tokenChartCacheRows)
+      .insert(result.toDto(id), mode: InsertMode.replace);
+
+  Future<void> remove(String id) =>
+      (_db.delete(_db.tokenChartCacheRows)..where((t) => t.id.equals(id))).go();
+
+  Future<void> clear() => _db.delete(_db.tokenChartCacheRows).go();
+}
+
+class TokenChartCacheRows extends Table with EntityMixin {
+  TextColumn get data => text()();
+}
+
+extension on List<TokenChartItem> {
+  TokenChartCacheRow toDto(String id) => TokenChartCacheRow(
+        id: id,
+        created: DateTime.now(),
+        data: json.encode(map((e) => e.toJson()).toList()),
```

Comment:
Don't use json encoding, in case of migration it will fail with an error which will be hard to catch.